uvicorn/protocols/websockets/websockets_impl.py

```diff
@@ -141,7 +141,9 @@ def connection_lost(self, exc: Optional[Exception]) -> None:
 
     def shutdown(self) -> None:
         self.ws_server.closing = True
-        self.transport.close()
+        task = asyncio.create_task(self.close(code=1012))
+        task.add_done_callback(self.on_task_complete)
+        self.tasks.add(task)
 
```

Comment:
Help needed here. 

`self.transport.close()` causes unexpected closure of connection at this place. However the transport is getting closed at a later stage.